src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataReadMeControl.xaml.cs

```diff
@@ -19,11 +15,84 @@ namespace NuGet.PackageManagement.UI
     /// <summary>
     /// Interaction logic for PackageMetadataReadMeControl.xaml
     /// </summary>
-    public partial class PackageMetadataReadMeControl : UserControl
+    public partial class PackageMetadataReadMeControl : UserControl, IDisposable
     {
+#pragma warning disable CS0618 // Type or member is obsolete
+        private IMarkdownPreview _markdownPreview;
+#pragma warning restore CS0618 // Type or member is obsolete
+        private bool _disposed = false;
+
+        [Obsolete]
         public PackageMetadataReadMeControl()
         {
             InitializeComponent();
         }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                _markdownPreview?.Dispose();
+                var viewModel = (ReadMePreviewViewModel)DataContext;
+                viewModel.PropertyChanged -= ViewModel_PropertyChangedAsync;
+            }
+
+            _disposed = true;
+        }
+
+        private void PackageMetadataReadMeControl_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            var viewModel = (ReadMePreviewViewModel)DataContext;
+            viewModel.PropertyChanged += ViewModel_PropertyChangedAsync;
+        }
+
+#pragma warning disable VSTHRD100 // Avoid async void methods
+        private async void ViewModel_PropertyChangedAsync(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+#pragma warning restore VSTHRD100 // Avoid async void methods
+        {
+            var viewModel = (ReadMePreviewViewModel)DataContext;
```

Comment:
Instead of casting the DataContext as the expected type whenever it's referenced, I like to have a private property that returns the DataContext as the expected type and use that instead.

Reply:
I like this :O